server/src/test/java/org/spine3/server/entity/EntityShould.java

```diff
@@ -185,11 +185,81 @@ public void return_id_protobuf_type_name() {
         assertEquals(expected, actual);
     }
 
-    private static Project newProject() {
-        final Project.Builder project = Project.newBuilder()
-                                               .setId(Given.AggregateId.newProjectId())
-                                               .setStatus(Project.Status.CREATED);
-        return project.build();
+    @Test
+    public void generate_non_zero_hash_code() {
+        final int hashCode = entityWithState.hashCode();
```

Comment:
In this case, a hash code is generated using an ID, state, etc. An ID is a non-empty String, therefore, the hash code of the entity cannot be zero. 
I guess it is better to rename a method to `generate_non_zero_hash_code_if_entity_has_non_empty_id_and_state`.